client/src/views/settings/Settings.vue

```diff
@@ -0,0 +1,162 @@
+<template>
+  <h1>Settings</h1>
+  <VRow>
+    <VCol md="6">
+      <VProgressCircular indeterminate v-if="loading"/>
+      <VAlert v-else-if="!!error"
+              color="error"
+      >
+        {{ error }}
+      </VAlert>
+      <div v-else>
+        <h2>General</h2>
+        <VForm @submit.prevent class="mt-4">
+          <AutosavingTextInput :on-submit="submit"
+                               :initial-value="settings?.eventName"
+                               name="eventName"
+                               label="Event name"
+                               input-type="text"
+                               setting-type="setting"
+                               :key="`eventName-${dataRefreshKey}`"
+          />
+          <AutosavingTextInput :on-submit="submit"
+                               :initial-value="settings?.eventTbaCode"
+                               name="eventTbaCode"
+                               label="Event TBA code"
+                               input-type="text"
+                               setting-type="setting"
+                               :key="`eventTbaCode-${dataRefreshKey}`"
+          />
+          <AutosavingTextInput :on-submit="submit"
+                               :initial-value="settings?.videoSearchDirectory"
+                               name="videoSearchDirectory"
+                               label="Video search directory"
+                               input-type="text"
+                               setting-type="setting"
+                               :key="`videoSearchDirectory-${dataRefreshKey}`"
+          />
+        </VForm>
+
+        <h2>YouTube</h2>
+
+        <h3 class="mb-2">OAuth2 client details</h3>
+        <VAlert class="mb-3"
+                v-if="!youTubeAuthState?.accessTokenStored"
+        >In your Google Cloud project, create an OAuth2 web client.<br />
+          <br />
+          Be sure to add <code>http://localhost:3000/api/v1/youtube/auth/callback</code> as an authorized redirect.
```

Comment:
Referencing `localhost` is fine here for now, but I could see a future where we'd want to populate that with whatever hostname was used in the browser to load the page. Or failing that, explicitly decide that the YouTube pairing process can only happen from the machine running the uploader but I'm not as much of a fan of that.

Reply:
It should probably pull the hostname/domain of the current page and use that as the suggestion. But technically you can already set it to whatever you please, although it requires manually adjusting the value in settings.json

Reply:
I fixed this up a little bit so it now uses `window.location.origin` which should give us everything before the path on the domain (e.g., "http://localhost:3000")

I also added a fancy copy button for the URL and the text input will select all its text when you focus it:
![image](https://github.com/gafirst/match-uploader/assets/5790137/d60b3eec-2d76-40bf-8d70-26feac468965)